File: src/0034.cpp
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        if(nums.size() < 1) {
            return {-1, -1};
        }
        bool found = false, failed = false;
        int first = 0, mid = 0;
        int last = nums.size() - 1;
        while(!found && !failed) {
            mid = (first + last) / 2;
            if(nums[mid] == target) {
                found = true;
            }else {
                if(first >= last) {
                    failed = true;
                }else {
                    if(nums[mid] > target) {
                        last = mid - 1;
                    }else {
                        first = mid + 1;
                    }
                }
            }
        }
        if(found) {
            int left = mid, right = mid;
            while(left >= 0 && nums[left] == target) {
                left--;
            }
            while(right < nums.size() && nums[right] == target) {
                right++;
            }
            return {++left, --right};
        }
        return {-1, -1};
    }
};

File: src/0034.py
class Solution:
    def searchRange(self, nums, target: int):
        if len(nums) < 1:
            return [-1, -1]
        found = searchfailed = False
        first = 0
        last = len(nums) - 1
        while not found and not searchfailed:
            middle = (first + last) // 2
            if nums[middle] == target:
                found = True
            else:
                if first >= last:
                    searchfailed = True
                else:
                    if nums[middle] > target:
                        last = middle - 1
                    else:
                        first = middle + 1
        if found:
            left = right = middle
            while left >= 0 and nums[left] == target:
                left -= 1
            while right < len(nums) and nums[right] == target:
                right += 1   
            return [left+1, right-1]
        else:
            return [-1, -1]
